multi-agent/lib/mas/elements/nodes/common/context_binder.py

```
@@ -0,0 +1,97 @@
+def bind_tool_context(
+    tools: List[BaseTool],
+    *,
+    session_id: str = "",
+    agent_id: str = "",
+) -> None:
+    """Inject runtime context into tools that expose ``bind_context()``."""
+    for tool in tools:
+        if hasattr(tool, "bind_context"):
+            try:
+                tool.bind_context(session_id=session_id, agent_id=agent_id)
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)